### PR TITLE
Update Material 3 `CircularProgressIndicator` for new visual style

### LINKED ISSUE #131690: Issue: Cannot theme progress indicators, many properties missing
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

## Steps to reproduce

The `ProgressIndicatorThemeData` is missing many key properties from `LinearProgressIndicator` and `CircularProgressIndicator` making it impossible to theme their styles.

### Expected results

Expect to be able to style the `LinearProgressIndicator` and `CircularProgressIndicator` with their `ProgressIndicatorThemeData` by also adding styles for properties:

- `borderRadius` The border radius of both the indicator and the linear track on a `LinearProgressIndicator`. https://github.com/flutter/flutter/blob/5d2c9d0933b9946d2eff8477edaceb9c91f61998/packages/flutter/lib/src/material/progress_indicator.dart#L291
- `strokeWidth`  The width of the line used to draw the circle on a `CircularProgressIndicator`. https://github.com/flutter/flutter/blob/5d2c9d0933b9946d2eff8477edaceb9c91f61998/packages/flutter/lib/src/material/progress_indicator.dart#L564
- `strokeAlign` The relative position of the stroke on a `CircularProgressIndicator`. https://github.com/flutter/flutter/blob/5d2c9d0933b9946d2eff8477edaceb9c91f61998/packages/flutter/lib/src/material/progress_indicator.dart#L565
- `strokeCap` The progress indicator's line ending on a `CircularProgressIndicator`. https://github.com/flutter/flutter/blob/5d2c9d0933b9946d2eff8477edaceb9c91f61998/packages/flutter/lib/src/material/progress_indicator.dart#L568

### Actual results

The newer `LinearProgressIndicator` and `CircularProgressIndicator` properties listed above that were added as enhancements, e.g. in [PR #122664](https://github.com/flutter/flutter/pull/122664) were forgotten or neglected to be added to `ProgressIndicatorThemeData` and are now missing, this is causing theming **issues**. 

The properties cannot be found in the latest version of `ProgressIndicatorThemeData`: 

https://github.com/flutter/flutter/blob/5d2c9d0933b9946d2eff8477edaceb9c91f61998/packages/flutter/lib/src/material/progress_indicator_theme.dart#L32

Therefore, we cannot theme the `LinearProgressIndicator` and `CircularProgressIndicator` to their desired style.

### Code sample

Code only via provided repo links showing the issue.

### Used Flutter version

Flutter version used to study the code and link to repo code samples:

Channel master, 3.13.0-15.0.pre.16

<details>
  <summary>Flutter doctor</summary>

```
flutter doctor -v
[✓] Flutter (Channel master, 3.13.0-15.0.pre.16, on macOS 13.4.1 22F770820d darwin-arm64,
    locale en-US)
    • Flutter version 3.13.0-15.0.pre.16 on channel master at
    • Framework revision fcd5a6c478 (3 hours ago), 2023-08-01 06:06:40 -0400
    • Engine revision 25b9d1088d
    • Dart version 3.2.0 (build 3.2.0-19.0.dev)
    • DevTools version 2.25.0
    • If those were intentional, you can disregard the above warnings; however it is
      recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/rydmike/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Community Edition (version 2023.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 74.0.4
    • Dart plugin version 231.9161.14

[✓] VS Code (version 1.79.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.4.1 22F770820d darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 115.0.5790.114

[✓] Network resources
    • All expected network resources are available.

```
</details>

### LINKED ISSUE #158106: Cannot override default `CircularProgressIndicator` size
### Use case

Currently, it's not possible to resize `CircularProgressIndicator`.

https://github.com/flutter/flutter/blob/8591d0c16a6c7edfbea3b5b1bf08ae09db117b71/packages/flutter/lib/src/material/progress_indicator.dart#L868-L871

### Proposal

Add ability to override `CircularProgressIndicator` size

### PR DESCRIPTION
**Pull request by TahaTesser**
Related [Update both `ProgressIndicator` for Material 3 redesign](https://github.com/flutter/flutter/issues/141340)
Fixes [Issue: Cannot theme progress indicators, many properties missing](https://github.com/flutter/flutter/issues/131690)
Fixes [Cannot override default `CircularProgressIndicator` size](https://github.com/flutter/flutter/issues/158106)

### Code sample

<details>
<summary>expand to view the code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
          // progressIndicatorTheme: const ProgressIndicatorThemeData(
          //   constraints: BoxConstraints.tightFor(width: 100, height: 100),
          //   strokeWidth: 12
          // ),
          ),
      home: Scaffold(
        appBar: AppBar(title: const Text('CircularProgressIndicator')),
        body: const Center(
          child: Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: [
              CircularProgressIndicator(year2023: false, value: 0.5),
              CircularProgressIndicator(year2023: false),
            ],
          ),
        ),
      ),
    );
  }
}
```

</details>

### Preview 

Screenshot 2024-11-04 at 16 01 57

### New custom `CircularProgressIndicator.constraints` and stroke  width

Screenshot 2024-11-04 at 16 02 40

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: dev/tools/gen_defaults/lib/progress_indicator_template.dart
Comment (navaronbracke) [inline dev/tools/gen_defaults/lib/progress_indicator_template.dart:19]: This does not have to be late?

File: packages/flutter/lib/src/material/progress_indicator.dart
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/progress_indicator.dart:613]: Can we put a brief comment here to indicate that the `if` is for the latest M3 appearance and the `else` part is for the `year2023` version? Just for better understanding for later maintenance:)

File: packages/flutter/lib/src/material/progress_indicator.dart
Comment (TahaTesser) [reply packages/flutter/lib/src/material/progress_indicator.dart:1333]: Earlier i tried 48x48 and found it to be too big so i compared it with Android components, it is 36 pixels with padding comes to 48x48. The actual shape is still 36x36. I can file an issue for incorrect specs.

File: packages/flutter/test/material/progress_indicator_test.dart
Comment (TahaTesser) [reply packages/flutter/test/material/progress_indicator_test.dart:1245]: https://github.com/flutter/flutter/pull/158104#discussion_r1828966498

File: packages/flutter/test/material/progress_indicator_test.dart
Comment (TahaTesser) [reply packages/flutter/test/material/progress_indicator_test.dart:1579]: https://github.com/flutter/flutter/pull/158104#discussion_r1828966498

File: packages/flutter/lib/src/material/progress_indicator.dart
Comment (TahaTesser) [reply packages/flutter/lib/src/material/progress_indicator.dart:945]: Briliant.

File: packages/flutter/lib/src/material/progress_indicator.dart
Comment (TahaTesser) [reply packages/flutter/lib/src/material/progress_indicator.dart:1338]: Filed an issue https://github.com/flutter/flutter/issues/158162

File: packages/flutter/lib/src/material/progress_indicator.dart
Comment (TahaTesser) [reply packages/flutter/lib/src/material/progress_indicator.dart:618]: Added a comment.

File: packages/flutter/lib/src/material/progress_indicator.dart
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/progress_indicator.dart:1335]: Ah got it. I just asked the Material team to see if they have any suggestions. Probably it's just Android team hasn't updated this component since I just feel the Material Design website should be the only source of truth:)

**Review by QuncCccccc — APPROVED** [review]
LGTM! Thanks for the update:)

_EDIT:_ I just triaged the goldens. I think we are good to go once the tests are fixed:)!

**Comment by TahaTesser** [comment]
Filed samples and migration guide PRs. To be landed once this PR lands.

**Comment by QuncCccccc** [comment]
> @QuncCccccc Looks like this is failing some google testings. Meanwhile I'm preparing sample and migration guide PRs.

Checking!

**Review by QuncCccccc — COMMENTED** [review]
Just checked the Google testing and I think the changes are all acceptable. So I updated the github status.

**Review by QuncCccccc — APPROVED** [review]
LGTM! Thanks so much for the update:)

_EDIT_: Oh seems Google testing status is removed so it shows the same failures again. Checking whether the failures are expected.

_EDIT_: Manually setup the Github status. This PR should be good to go:)

**Pull request merged**
